### ISSUE TITLE
unique parameter ignored in a certain case in the default implementation of the taskWithFilenameBuilder

### ISSUE THREAD

**Issue by SzecsenyiPeter** [781flyingdutchman/background_downloader#267]
If I create a DownloadTask with the default implementation of the _taskWithFilenameBuilder_, and set the unique parameter to false, it still ends up generating a unique name.  Example of usage:
```
    DownloadTask task = await DownloadTask(
        url: url,
        baseDirectory: BaseDirectory.applicationSupport,
    )
        .withSuggestedFilename(unique: false);
```
I tracked down the issue to the 131. line of the _lib/src/util_ file:
`return uniqueFilename(task.copyWith(filename: suggestedFilename), true);`
I copied the default implementation of the _taskWithFilenameBuilder_, and I changed the second parameter of this call to the _unique_ parameter of the encosing method.  This solved the problem.
Am I missing some explicit reason why the _unique_ parameter is always set to true in this case?

**Comment by 781flyingdutchman** [comment]
Thank you, that's a solid bug, that was apparently not covered in the tests. I'll apply your fix in the next release - thanks for sharing both the issue and the solution!

**Comment by 781flyingdutchman** [comment]
Fixed in V8.4.0. Thanks again for your help!
